### DISCUSSION TITLE
Rust question about dereferencing

### DISCUSSION THREAD

**Discussion by JasonLi-cn** [databendlabs/databend#5032] [Q&A] [answered]
https://github.com/datafuselabs/databend/blob/5c370f5c8861bde6270d1e8f59ed2e8e9822cacf/query/src/pipelines/transforms/group_by/aggregator.rs#L93-L103

Why does dereferencing bring benefits？
"The number of times that Deref::deref needs to be inserted is resolved at compile time, so there is no runtime penalty for taking advantage of deref coercion!" in https://doc.rust-lang.org/book/ch15-02-deref.html#implicit-deref-coercions-with-functions-and-methods

**Comment by zhang2014** [comment, marked as answer]
Obsolete comments. This is because when we use loop match, "self.field" Dereference causes the compiler to optimize loop match to match loop

**Comment by zhang2014** [comment]
https://github.com/datafuselabs/databend/pull/5042
